### PR TITLE
Implement SBT signaling standard

### LINKED ISSUE #9: SBT signaling standard
hi,

I came across your Guild Pin product and thought it was really cool. I‘m the author of https://eips.ethereum.org/EIPS/eip-5192 and we suggest to implement events for locking and unlocking tokens, as well as a locked function to get the status of a token‘s lock.

From what I can tell, your implementation reverts on all transferring functions, which is also what we suggest doing in the standard.

Although our standard allows unlocking tokens again, you don‘t have to implement that. 

Our main idea is that by emitting the Lock and Unlock events and through the interface identifier, we can let wallets know when a token is soulbound, such that they can hide the transfer button, and inform the user about non-transferability.

E.g. OpenSea added partial support for this and the more people we can convince of adding these two events and the view function, the more the value if SBTs can grow 🤞

Hope I could sell you on this standard! It‘s far from perfect, but it could help grow the space a little!

IMG_3985

happy to answer any questions or even help implementing! Just text me

best,
Tim

### PR DESCRIPTION
**Pull request by TomiOhl**
Resolves #9

### CODE REVIEW

File: contracts/token/SoulboundERC721.sol [hunk 1]
```diff
@@ -33,10 +34,15 @@ contract SoulboundERC721 is ERC721Upgradeable, ERC721EnumerableUpgradeable, IERC
     ) public view virtual override(ERC721EnumerableUpgradeable, ERC721Upgradeable, IERC165Upgradeable) returns (bool) {
         return
             interfaceId == 0x49064906 || // ERC4906
+            interfaceId == type(IERC5192).interfaceId ||
             interfaceId == type(IERC721EnumerableUpgradeable).interfaceId ||
             super.supportsInterface(interfaceId);
     }
 
+    function locked(uint256) external pure returns (bool) {
```
Comment (TimDaub) [inline contracts/token/SoulboundERC721.sol:42]: According to the spec, for all tokens that don‘t exist („all tokens assigned to the zero address“) the locked function must throw.


Anyways, there also exist implementations where we forgot this - but maybe it‘s good if we strictly go by the spec.

**Comment by TimDaub** [comment]
Thank you for actually implementing this! I looked over it and I think it is all good except for the one comment I left.

You are not planning to ever unlock the tokens right? And claim is the only function that mints the token to a receiver? Does burning work?

File: contracts/token/SoulboundERC721.sol (hunk 1 above)
Comment (TomiOhl) [reply contracts/token/SoulboundERC721.sol]: Fair enough. Implemented in 357e8df

**Comment by TomiOhl** [comment]
> Thank you for actually implementing this! I looked over it and I think it is all good except for the one comment I left.
> 
> You are not planning to ever unlock the tokens right? And claim is the only function that mints the token to a receiver? Does burning work?

Yes, the tokens will never be unlocked. Claim is the only function that can mint the token. Burning works.

**Comment by TomiOhl** [comment]
Gonna go ahead and merge this as the review comment is addressed and I've read through the spec one more time.
Thanks again TimDaub!

**Pull request merged**